recipes/libtool/all/conanfile.py

```diff
@@ -26,106 +33,109 @@ class LibtoolConan(ConanFile):
         "fPIC": True,
     }
 
-    exports_sources = "patches/**"
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def export_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("automake/1.16.5")
+        #self.requires("m4/1.4.19") TODO: add as runtime dependency
```

Comment:
I'm clarifying the comment - I can see in the libtool/libtoolize scripts that `m4` is used directly, but I'm not sure whether it is generally used or just when debugging. It is either way a transitive dependency by way of automake. So while it may be "more correct" to express the dependency here - it may be "too strict" given that all versions are fixed, so adding it may cause version conflicts - it will be easier dealt with once we start expressing dependencies with version ranges.